### ISSUE TITLE
java.lang.IndexOutOfBoundsException: bitIndex < 0: -1

### ISSUE THREAD

**Issue by jordansissel** [elastic/logstash#6830]
```
% bin/logstash -e 'input { generator { threads => 2 }  } output { null { } }' 
Sending Logstash's logs to /home/jls/build/logstash-5.3.0-SNAPSHOT/logs which is now configured via log4j2.properties
[2017-03-22T15:50:32,638][INFO ][logstash.setting.writabledirectory] Creating directory {:setting=>"path.queue", :path=>"/home/jls/build/logstash-5.3.0-SNAPSHOT/data/queue"}
[2017-03-22T15:50:32,769][INFO ][logstash.pipeline        ] Starting pipeline {"id"=>"main", "pipeline.workers"=>8, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>5, "pipeline.max_inflight"=>1000}
[2017-03-22T15:50:32,831][INFO ][logstash.pipeline        ] Pipeline main started
[2017-03-22T15:50:32,899][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
Exception in thread "[main]>worker5" java.lang.IndexOutOfBoundsException: bitIndex < 0: -1
	at java.util.BitSet.set(BitSet.java:444)
	at org.logstash.ackedqueue.Page.ack(Page.java:104)
	at org.logstash.ackedqueue.Queue.ack(Queue.java:552)
	at org.logstash.ackedqueue.Batch.close(Batch.java:26)
	at org.logstash.ackedqueue.ext.JrubyAckedBatchExtLibrary$RubyAckedBatch.ruby_close(JrubyAckedBatchExtLibrary.java:80)
	at org.logstash.ackedqueue.ext.JrubyAckedBatchExtLibrary$RubyAckedBatch$INVOKER$i$0$0$ruby_close.call(JrubyAckedBatchExtLibrary$RubyAckedBatch$INVOKER$i$0$0$ruby_close.gen)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:134)
	at org.jruby.ast.CallNoArgNode.interpret(CallNoArgNode.java:60)
	at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
	at org.jruby.ast.BlockNode.interpret(BlockNode.java:71)
	at org.jruby.evaluator.ASTInterpreter.INTERPRET_METHOD(ASTInterpreter.java:74)
	at org.jruby.internal.runtime.methods.InterpretedMethod.call(InterpretedMethod.java:139)
....
```

Confirmed on Logstash 5.2.1,  5.3.0-SNAPSHOT, and 5.4.0-SNAPSHOT.

Logstash yaml (Reader's note, please do not use this configuration with such small pages in production. I am using it to find bugs and test edge cases)

```
queue.type: persisted
queue.page_capacity: 1.234mb
queue.max_bytes: 10.1234mb
```

**Comment by original-brownbear** [comment]
@jordansissel tried reproducing this in `master`, seems fixed to me :)

**Comment by original-brownbear** [comment]
@jordansissel this is fixed for sure now I think. I could eventually reproduce this one as a result of #7382 which is fixed now. Fixed in 5.4 and newer now :)
